File: Practica1/Backend-Nodejs/src/repositories/playlist.repository.js
const pool = require("../config/db");

async function findMovieById(id_pelicula) {
  const result = await pool.query(
    "SELECT * FROM peliculas WHERE id_pelicula = $1",
    [id_pelicula]
  );
  return result.rows[0];
}

async function addToPlaylist(id_usuario, id_pelicula) {
  const result = await pool.query(
    `
    INSERT INTO lista_reproduccion (id_usuario, id_pelicula)
    VALUES ($1, $2)
    RETURNING *
    `,
    [id_usuario, id_pelicula]
  );
  return result.rows[0];
}

async function getUserPlaylist(id_usuario) {
  const result = await pool.query(
    `
    SELECT p.*
    FROM lista_reproduccion lr
    JOIN peliculas p ON lr.id_pelicula = p.id_pelicula
    WHERE lr.id_usuario = $1
    ORDER BY lr.fecha_agregado DESC
    `,
    [id_usuario]
  );
  return result.rows;
}

async function removeFromPlaylist(id_usuario, id_pelicula) {
  await pool.query(
    `
    DELETE FROM lista_reproduccion
    WHERE id_usuario = $1 AND id_pelicula = $2
    `,
    [id_usuario, id_pelicula]
  );
}

module.exports = {
  findMovieById,
  addToPlaylist,
  getUserPlaylist,
  removeFromPlaylist
};

File: Practica1/Backend-python/app/repositories/playlist_repository.py
from app.config import get_connection
import psycopg2

def find_movie_by_id(id_pelicula):
    conn = get_connection()
    cursor = conn.cursor()

    cursor.execute(
        "SELECT * FROM peliculas WHERE id_pelicula = %s",
        (id_pelicula,)
    )

    movie = cursor.fetchone()

    cursor.close()
    conn.close()

    return movie


def add_to_playlist(id_usuario, id_pelicula):
    conn = get_connection()
    cursor = conn.cursor()

    cursor.execute(
        """
        INSERT INTO lista_reproduccion (id_usuario, id_pelicula)
        VALUES (%s, %s)
        RETURNING *
        """,
        (id_usuario, id_pelicula)
    )

    conn.commit()
    result = cursor.fetchone()

    cursor.close()
    conn.close()

    return result


def get_user_playlist(id_usuario):
    conn = get_connection()
    cursor = conn.cursor()

    cursor.execute(
        """
        SELECT p.id_pelicula, p.titulo, p.director,
               p.anio_estreno, p.url_contenido,
               p.poster, p.estado
        FROM lista_reproduccion lr
        JOIN peliculas p ON lr.id_pelicula = p.id_pelicula
        WHERE lr.id_usuario = %s
        ORDER BY lr.fecha_agregado DESC
        """,
        (id_usuario,)
    )

    movies = cursor.fetchall()

    cursor.close()
    conn.close()

    return movies


def remove_from_playlist(id_usuario, id_pelicula):
    conn = get_connection()
    cursor = conn.cursor()

    cursor.execute(
        """
        DELETE FROM lista_reproduccion
        WHERE id_usuario = %s AND id_pelicula = %s
        """,
        (id_usuario, id_pelicula)
    )

    conn.commit()

    cursor.close()
    conn.close()